test/health_checker_unittest.cc

```diff
@@ -0,0 +1,72 @@
+#include "../src/health_checker.h"
+#include "gtest/gtest.h"
+#include <stdio.h>
+#include <sstream>
+#include <boost/filesystem.hpp>
+
+
+namespace google {
+
+class HealthCheckerUnittest : public ::testing::Test {
+ protected:
+  static void Cleanup(HealthChecker* health_checker){
```

Comment:
Why didn't you just use the gtest `TearDown()` here?

Reply:
Because the `health_checker` object has to be constructed too late to be a class member. Unless you can think of a way to propagate `test_info_->name()` from a subclass into the `HealthCheckUnittest` constructor...

Reply:
SGTM